### ISSUE TITLE
Empty object is returned when using React Navigation Drawer

### ISSUE THREAD

**Issue by josiext** [realm/realm-js#4734]
### How frequently does the bug occur?

All the time

### Description

If we init a react native app using CLI and install react navigation drawer and stack and put this in App.tsx you will be able to see the issue.

```
import React, {useEffect} from 'react';
import {Text, SafeAreaView} from 'react-native';
import {GestureHandlerRootView} from 'react-native-gesture-handler';
import {NavigationContainer} from '@react-navigation/native';
import Realm, {BSON} from 'realm';

import {createStackNavigator} from '@react-navigation/stack';
import {createDrawerNavigator} from '@react-navigation/drawer';

const Drawer = createDrawerNavigator();

const Cat = {
  name: 'Cat',
  properties: {
    _id: 'objectId',
    name: 'string',
    age: 'int',
    type: 'string',
  },
};

Realm.open({
  schema: [Cat],
}).then(realm => {
  let aliceCat;
  realm.write(() => {
    aliceCat = realm.create('Cat', {
      _id: new BSON.ObjectID(),
      name: 'Alice',
      age: 14,
      type: 'Calico',
    });
  });

  const cats = realm.objects('Cat');
 
 console.log(cats); // -> empty when using Drawer
});

function Home() {
  return <Text style={{color: 'red'}}>Home</Text>;
}

function MyDrawer() {
  return (
    <Drawer.Navigator>
      <Drawer.Screen name="Feed" component={Home} />
    </Drawer.Navigator>
  );
}

const Stack = createStackNavigator();

function MyStack() {
  return (
    <Stack.Navigator initialRouteName="Drawer">
      <Stack.Screen name="Home" component={Home} />
      <Stack.Screen name="Drawer" component={MyDrawer} />
    </Stack.Navigator>
  );
}

const App = () => {
  return (
    <GestureHandlerRootView style={{flex: 1}}>
      <SafeAreaView style={{flex: 1}}>
        <NavigationContainer>
          <MyStack />
        </NavigationContainer>
      </SafeAreaView>
    </GestureHandlerRootView>
  );
};

export default App;
```

But if remove the drawer the issue is gone and you can get the data from Realm:

```
import React, {useEffect} from 'react';
import {Text, SafeAreaView} from 'react-native';
import {GestureHandlerRootView} from 'react-native-gesture-handler';
import {NavigationContainer} from '@react-navigation/native';
import Realm, {BSON} from 'realm';

import {createStackNavigator} from '@react-navigation/stack';
import {createDrawerNavigator} from '@react-navigation/drawer';

/* const Drawer = createDrawerNavigator();
 */
const Cat = {
  name: 'Cat',
  properties: {
    _id: 'objectId',
    name: 'string',
    age: 'int',
    type: 'string',
  },
};

Realm.open({
  schema: [Cat],
}).then(realm => {
  let aliceCat;
  realm.write(() => {
    aliceCat = realm.create('Cat', {
      _id: new BSON.ObjectID(),
      name: 'Alice',
      age: 14,
      type: 'Calico',
    });
  });

  console.log(aliceCat);

  const cats = realm.objects('Cat');
  console.log(cats);
});

function Home() {
  useEffect(() => {
    load();
  }, []);

  const load = () => {};

  console.log('HOME');
  return <Text style={{color: 'red'}}>Home</Text>;
}

/* function MyDrawer() {
  return (
    <Drawer.Navigator>
      <Drawer.Screen name="Feed" component={Home} />
    </Drawer.Navigator>
  );
} */

const Stack = createStackNavigator();

function MyStack() {
  return (
    <Stack.Navigator initialRouteName="Home">
      <Stack.Screen name="Home" component={Home} />
      {/*       <Stack.Screen name="Drawer" component={MyDrawer} />
       */}
    </Stack.Navigator>
  );
}

const App = () => {
  return (
    <GestureHandlerRootView style={{flex: 1}}>
      <SafeAreaView style={{flex: 1}}>
        <NavigationContainer>
          <MyStack />
        </NavigationContainer>
      </SafeAreaView>
    </GestureHandlerRootView>
  );
};

export default App;
```

**dependencies:** 

```
"@react-navigation/drawer": "6.4.3",
"@react-navigation/native": "6.0.11",
"@react-navigation/stack": "6.2.2",
"react": "18.0.0",
"react-native": "0.69.1",
"react-native-gesture-handler": "2.5.0",
"react-native-reanimated": "2.9.1",
"react-native-safe-area-context": "4.3.1",
"react-native-screens": "3.15.0",
"realm": "10.19.5"
```

### Can you reproduce the bug?

Yes, always

### Version

10.19.5

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

Yes, using encryption

### Platform OS and version(s)

Android 11

### Build environment

Which debugger for React Native: ..

**Comment by takameyer** [comment]
@josiext I noticed you have `react-native-reanimated` v2 as a dependency.  Our current v10 of realm has compatibility issues with this.  I welcome you to try our v11 release candidate to verify that the problem goes away.

**Comment by JoseNunezBV** [comment]
> @josiext I noticed you have `react-native-reanimated` v2 as a dependency. Our current v10 of realm has compatibility issues with this, specifically on Android builds. I welcome you to try our v11 release candidate to verify that the problem goes away.
> 
> ```
> npm install realm@11.0.0-rc.1
> ```

Great! We solved it installing that Realm version. Thanks!

**Comment by kneth** [comment]
As it sounds like it can been solved, I close the issue. Don't hesitate to either reopen or create a new issue if the problem still is observed.

**Comment by ivanhoe-dev** [comment]
I also have the same issue and resolved by upgrading to  `11.0.0-rc.1`. For those who have the same issue, please note that `11.0.0-rc.0` not working, only `11.0.0-rc.1` works
